### ISSUE TITLE
Module version is not treeshakable and has side-effects on import

### ISSUE THREAD

**Issue by maranomynet** [davidjbradshaw/iframe-resizer#1155]
if you import either `contentWindow` or `iframeResize` from the `'iframe-resizer'` module, for example like so

```js
import { iframeResize } from 'iframe-resizer';
```

…you'd expect the reuslting browser bundle to only contain the `iframeResize` function.

What happens, however, is that both the host script and the contentWindow script get bundled (and executed!).

This result is completely un-obvious, until you open and carefully inspect the output bundle (which most people never do), or if you inspect the window of the webpage running the bundled code and notice that it has both the client and host scripts running at the same time.

This happens beacuse both modules (`src/iframeResizer.js`, and `src/iframeResizer.contentWindow.js`) contain an IIFE with immediate side-effects (creating global variables, adding window event hanlers, etc.) and thus the code can't be safely excluded from the output bundle.

...

There exist a few ways to resolve this.

I guess the simplest would be to stop exporting both modules from `index.js`, and instead tell people they must import `from 'iframe-resizer/js/iframeResizer.js'` and `from 'iframe-resizer/js/iframeResizer.contentWindow.js'`.

An even better solution would be to refactor the source files to export an `init` function, and then create separate output files for browsers and node.js import usage.

The browser files would then immediately invoke their respective `init()` function, whereas the node.js module would simply export the init functions and have developers decide when and if to run them.

**Comment by maranomynet** [comment]
BTW, my current workaround is to import the host function like this

```js
import iframeResizer from 'iframe-resizer/js/iframeResizer.js';
```

which solves the problem, I guess, but only as long as the script path doesn't change.

A formal documentation of this would be nice.

**Comment by alexcumplido** [comment]
Hi @SamMousa  trying to implement your approach but I have a few questions.

- Do you still placing a script in the external page containing the frame as specified in the docs (https://github.com/davidjbradshaw/iframe-resizer#usage)
-
- How are you using the `<iframe/>` element on svelte component?

**Comment by SamMousa** [comment]
Yes, external page is a classic php monolith that renders html in which we include the script.

The svelte app is deployed separately so it has a URL

**Comment by davidjbradshaw** [comment]
For version 5 I am breaking the library up into separate modules. If your interested the draft docs are now online at https://iframe-resizer.com and their is an early alpha release on NPM. Would be interested in any feedback you have.

**Comment by davidjbradshaw** [comment]
Fixed in V5.0.0

Please see https://iframe-resizer.com
